setup

```diff
@@ -39,11 +65,25 @@ download_jar() {
   prefix="http://search.maven.org/remotecontent?filepath="
   path=${1//.//}
   jar_name=$artifact_id-$version.jar
+  tmp_dest=$(mktemp -u)
+  jar_dest=$dependencies_dir/$jar_name
   url=${prefix}${path}/${artifact_id}/${version}/${jar_name}
-  [[ -f ${dependencies_dir}/${jar_name} ]] || wget -P ${dependencies_dir} $url
+
+  # file exists? then don't download it!
+  [[ -f ${jar_dest} ]] && return
+
+  # download the file to a temp location, then if move it here if successful
+  echo "Downloading $jar_name ($url)..."
+  wget -q -P ${dependencies_dir} $url -O ${tmp_dest} && \
+    mv ${tmp_dest} ${jar_dest}
+
+  # Purge our temp file regardless of the status here
+  rm -f ${tmp_dest}
```

Comment:
What's the advantages here by downloading the jar to temp place and moving it over?